### PR TITLE
[bug 1135596] Generic error dialog

### PR DESCRIPTION
**Pull request by rehandalal**
r?

### CODE REVIEW

File: app/js/sumo_db.js [hunk 1]
```diff
@@ -12,6 +12,26 @@
   var sequence_id = 0;
   var last_request;
 
+  function show_generic_error() {
```
Comment (Rik) [inline app/js/sumo_db.js]: I'd like to move this into another file to make sure sumo_db only does HTTP requests and no UI. Can you create an event and dispatch it ([docs for modern way to that](https://developer.mozilla.org/en-US/docs/Web/API/Document/createEvent))?
And then listen for this event in a new file. I suggest calling this event network-error.

File: app/js/sumo_db.js [hunk 2]
```diff
@@ -12,6 +12,26 @@
...
+    var generic_error = document.getElementById('generic_error');
+
+    if (!generic_error) {
+      var html = nunjucks.render('generic_error.html');
+      document.body.insertAdjacentHTML('beforeend', html);
+
+      generic_error = document.getElementById('generic_error');
+      generic_error.addEventListener('click', function(evt) {
+        if (!evt.target.classList.contains('recommend')) {
```
Comment (Rik) [inline app/js/sumo_db.js]: Instead of checking for this class, you can listen to the `submit` event.

File: app/js/sumo_db.js [hunk 3]
```diff
@@ -12,6 +12,13 @@
   var sequence_id = 0;
   var last_request;
 
+  function trigger_error() {
+    var event = new Event('network-error');
+    document.dispatchEvent(event);
+  }
+
+
+
```
Comment (Rik) [inline app/js/sumo_db.js]: Nit: Blank lines

Comment (rehandalal) [inline app/js/sumo_db.js]: gah!

File: app/views/generic_error.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<form id="generic_error" role="dialog" data-type="confirm">
+  <section>
+    <h1>Whoops!</h1>
+    <p>There was a problem with Buddy Up.</p>
+  </section>
+  <menu>
+    <button class="recommend">Ok</button>
```
Comment (Rik) [inline app/views/generic_error.html]: ALL_CAPS !!!1!! OK

File: app/kb.html [hunk 5]
```diff
@@ -8,6 +8,7 @@
 
     <link rel="stylesheet" href="css/building_blocks/layout.css" media="screen" />
     <link rel="stylesheet" href="css/building_blocks/headers.css" media="screen" />
+    <link rel="stylesheet" href="css/building_blocks/confirm.css" media="screen" />
```
Comment (Rik) [inline app/kb.html]: This is missing error_controller.js

**Comment by rehandalal** [comment]
https://github.com/mozilla/buddyup/commit/1d27a49bc6d2326da35c8c6c9644687e275cae83

**Pull request closed without merging**